src/cascadia/TerminalApp/TmuxControl.cpp

```diff
@@ -0,0 +1,1480 @@
+    }
+
+    void TmuxControl::_SendKey(int paneId, const std::wstring keys)
+    {
+        auto cmd = std::make_unique<SendKey>();
+        cmd->paneId = paneId;
+        cmd->keys = keys;
+
+        _SendCommand(std::move(cmd));
+        _ScheduleCommand();
+    }
+
+    std::wstring TmuxControl::SendKey::GetCommand()
+    {
+        std::wstring out = L"";
+        for (auto & c : this->keys)
+        {
+            out += std::format(L"{:#x} ", c);
+        }
+
+        return std::wstring(std::format(L"send-key -t %{} {}\n", this->paneId, out));
+    }
+
+
+    void TmuxControl::_SetOption(const std::wstring& option)
+    {
+        auto cmd = std::make_unique<SetOption>();
+        cmd->option = option;
+        _SendCommand(std::move(cmd));
+        _ScheduleCommand();
+    }
+
+    std::wstring TmuxControl::SetOption::GetCommand()
+    {
+        return std::wstring(std::format(L"set-option {}\n", this->option));
+    }
+
+    void TmuxControl::_SplitPane(std::shared_ptr<Pane> pane, SplitDirection direction)
+    {
+        if (_splittingPane.first != nullptr)
+        {
+            return;
+        }
+
+        if (!pane)
+        {
+            return;
+        }
+
+        int paneId = -1;
+        for (auto& p : _attachedPanes)
+        {
+            if (pane->GetTerminalControl() == p.second.control)
+            {
+                paneId = p.first;
+            }
+        }
+
+        if (paneId == -1)
+        {
+            return;
+        }
+
+        _splittingPane = {pane, direction};
+        auto cmd = std::make_unique<struct SplitPane>();
+        cmd->direction = direction;
+        cmd->paneId = paneId;
+        _SendCommand(std::move(cmd));
+        _ScheduleCommand();
+    }
+
+    std::wstring TmuxControl::SplitPane::GetCommand()
+    {
+        if (this->direction == SplitDirection::Right)
+        {
+            return std::wstring(std::format(L"split-window -h -t %{}\n", this->paneId));
+        }
+        else
+        {
+            return std::wstring(std::format(L"split-window -v -t %{}\n", this->paneId));
+        }
+    }
+
+    // From controller to tmux
+    void TmuxControl::_CommandHandler(const std::wstring& result)
+    {
+        if (_cmdState == WAITING && _cmdQueue.size() > 0)
+        {
+            auto cmd = _cmdQueue.front().get();
+            cmd->ResultHandler(result, *this);
+            _cmdQueue.pop_front();
+            _cmdState = READY;
+        }
+    }
+
+    void TmuxControl::_SendCommand(std::unique_ptr<TmuxControl::Command> cmd)
+    {
+        _cmdQueue.push_back(std::move(cmd));
+    }
+
+    void TmuxControl::_ScheduleCommand()
+    {
+        if (_cmdState != READY)
+        {
+            return;
+        }
+
+        if (_cmdQueue.size() > 0)
+        {
+            _cmdState = WAITING;
+
+            auto cmd = _cmdQueue.front().get();
+            auto cmdStr = cmd->GetCommand();
+            _control.RawWriteString(cmdStr);
+        }
+    }
```

Comment:
I didn't understand why you made sure to only ever send 1 command at a time. I removed this in my branch, and it seems to work fine. It did cause various issues, however. One of the problems it caused was that startup worked faster now and that resulted in a race with pane creation IIRC. I believe I fixed it, however.